src/web/routes/application/postcode/postcode.test.js

```diff
@@ -0,0 +1,76 @@
+const test = require('tape')
+const sinon = require('sinon')
+const proxyquire = require('proxyquire')
+const TEST_FIXTURES = require('./test-fixtures.json')
+
+const request = sinon.stub()
+const { behaviourForPost } = proxyquire('./postcode', { 'request-promise': request })
+const config = { environment: { OS_PLACES_API_KEY: '123' } }
+
+test('behaviourForPost() handles succesful address lookup', async (t) => {
+  const req = { body: { postcode: 'BS7 8EE' }, session: {} }
+  const res = {}
+  const next = sinon.spy()
+
+  const expectedAdresses = [
+    {
+      ADDRESS: 'ALAN JEFFERY ENGINEERING, 1, VALLEY ROAD, PLYMOUTH, PL7 1RF',
+      ORGANISATION_NAME: 'ALAN JEFFERY ENGINEERING',
+      BUILDING_NUMBER: '1',
+      THOROUGHFARE_NAME: 'VALLEY ROAD',
+      POST_TOWN: 'PLYMOUTH',
+      POSTCODE: 'PL7 1RF',
+      LOCAL_CUSTODIAN_CODE_DESCRIPTION: 'CITY OF PLYMOUTH'
+    }, {
+      ADDRESS: 'DULUX DECORATOR CENTRE, 2, VALLEY ROAD, PLYMOUTH, PL7 1RF',
+      ORGANISATION_NAME: 'DULUX DECORATOR CENTRE',
+      BUILDING_NUMBER: '2',
+      THOROUGHFARE_NAME: 'VALLEY ROAD',
+      POST_TOWN: 'PLYMOUTH',
+      POSTCODE: 'PL7 1RF',
+      LOCAL_CUSTODIAN_CODE_DESCRIPTION: 'CITY OF PLYMOUTH'
+    }, {
+      ADDRESS: 'MILL AUTOS, 3, VALLEY ROAD, PLYMOUTH, PL7 1RF',
+      ORGANISATION_NAME: 'MILL AUTOS',
+      BUILDING_NUMBER: '3',
+      THOROUGHFARE_NAME: 'VALLEY ROAD',
+      POST_TOWN: 'PLYMOUTH',
+      POSTCODE: 'PL7 1RF',
+      LOCAL_CUSTODIAN_CODE_DESCRIPTION: 'CITY OF PLYMOUTH'
+    }
+  ]
+
+  request.returns(Promise.resolve(TEST_FIXTURES))
+
+  try {
+    await behaviourForPost(config)(req, res, next)
+
+    t.deepEqual(req.session.postcodeLookupResults, expectedAdresses, 'adds postcode lookup results to session')
+    t.equal(next.called, true, 'calls next()')
+  } catch (error) {
```

Comment:
We don't normally wrap tests in a try/catch, wouldn't it automatically fail the test if it threw an error?

Reply:
It's not strictly necessary, but I've added a comment about why I've added this. I can remove the try / catch if people prefer.